core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1018,70 +1016,18 @@ private[spark] object Utils extends Logging {
     " " + (System.currentTimeMillis - startTimeMs) + " ms"
   }
 
-  private def listFilesSafely(file: File): Seq[File] = {
-    if (file.exists()) {
-      val files = file.listFiles()
-      if (files == null) {
-        throw new IOException("Failed to list files for dir: " + file)
-      }
-      files
-    } else {
-      List()
-    }
-  }
-
-  /**
-   * Lists files recursively.
-   */
-  def recursiveList(f: File): Array[File] = {
```

Comment:
Yes this is only used by test suites for now, but can you still keep this in `Utils` so we don't have to move it back when we want to use it in the production code (I assume this may happen someday).

Reply:
I disagree that we should keep things in production code "just in case".

This is also an example of a function that if used without care is dangerous. `Files.walk` is a much better way of achieving this.